test/test_parentwork.py

```diff
@@ -85,14 +134,72 @@ def test_no_force(self):
     @unittest.skipUnless(
         os.environ.get('INTEGRATION_TEST', '0') == '1',
         'integration testing not enabled')
-    def test_direct_parent_work(self):
+    def test_direct_parent_work_real(self):
         mb_workid = u'2e4a3668-458d-3b2a-8be2-0b08e0d8243a'
         self.assertEqual(u'f04b42df-7251-4d86-a5ee-67cfa49580d1',
                          parentwork.direct_parent_id(mb_workid)[0])
         self.assertEqual(u'45afb3b2-18ac-4187-bc72-beb1b1c194ba',
                          parentwork.work_parent_id(mb_workid)[0])
 
 
+class ParentWorkTest(unittest.TestCase, TestHelper):
+    def setUp(self):
+        """Set up configuration"""
+        self.setup_beets()
+        self.load_plugins('parentwork')
+        #self.patcher1 = unittest.mock.patch('musicbrainzngs.get_work_by_id')
+        #self.mock_workid_response = self.patcher1.start()
+        musicbrainzngs.get_work_by_id = Mock(side_effect=mock_workid_response)
```

Comment:
This is close! The only trouble is that this will be _permanent_. That is, it's changing what `musicbrainzngs.get_work_by_id` points to when the test gets set up, and it never changes it back when the test is finished. So any tests that run after this will see the mock instead of the original function.

I think you want this:

```py
self.patcher = patch('musicbrainzngs.get_work_by_id', side_effect=mock_workid_response)
```

And then, in `tearDown`:

```py
self.patcher.stop()
```

to clean up.

Reply:
Done!